src/main/java/frc/robot/subsystems/swerve/OdometryThreadIO.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems.swerve;
+
+import frc.robot.subsystems.swerve.PhoenixOdometryThread.Samples;
+import frc.robot.subsystems.swerve.PhoenixOdometryThread.SignalID;
+import frc.robot.subsystems.swerve.PhoenixOdometryThread.SignalType;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.littletonrobotics.junction.LogTable;
+import org.littletonrobotics.junction.inputs.LoggableInputs;
+
+public interface OdometryThreadIO {
+  public class OdometryThreadIOInputs implements LoggableInputs {
+    List<Samples> sampledStates = List.of();
+
+    @Override
+    public void toLog(LogTable table) {
+      var timestamps = sampledStates.stream().mapToDouble(Samples::timestamp).toArray();
+      Set<Integer> modIds = new HashSet();
+      for (int i = 0; i < sampledStates.size(); i++) {
+        var sample = sampledStates.get(i);
+        for (var signal : sample.values().entrySet()) {
+          table.put(
+              "Data/" + i + " " + signal.getKey().type().toString() + " " + signal.getKey().modID(),
```

Comment:
im using the index so that each timestamp has some identifier to get all the timestamps at that system. originally i put the timestamp in the key, but that means a new NT topic is created for each timestamp sample which floods ascope with topics. I like having the index before the modID so that all the values from one timestamp are next to each other for comparison for debugging. I think the order should be preserved since this order comes from `journal`s order in `PhoenixOdometryThread`, which I think should be constant since the thread is always sampled in time order? that contract isnt explicit at the moment though (although the only other user, the sim on the maple-swerve-sim branch, also maintains it)